packages/database/src/core/PersistentConnection.ts

```diff
@@ -184,6 +193,57 @@ export class PersistentConnection extends ServerActions {
     }
   }
 
+  get(query: Query): Promise<string> {
+    const deferred = new Deferred<string>();
+    const request = {
+      p: query.path.toString(),
+      q: query.queryObject()
+    };
+    const outstandingGet = {
+      action: 'g',
+      request,
+      onComplete: (message: { [k: string]: unknown }) => {
+        const payload = message['d'] as string;
+        if (message['s'] === 'ok') {
+          this.onDataUpdate_(
+            request['p'],
+            payload,
+            /*isMerge*/ false,
+            /*tag*/ null
+          );
+          deferred.resolve(payload);
+        } else {
+          deferred.reject(payload);
+        }
+      }
+    };
+    this.outstandingGets_.push(outstandingGet);
+    this.outstandingGetCount_++;
+    const index = this.outstandingGets_.length - 1;
+
+    if (!this.connected_) {
+      setTimeout(() => {
```

Comment:
We should consider cancelling this timeout if the `get` succeeded (since it may keep the Node process running). We can do this in a follow-up.

Reply:
Will do this in a followup.